kotlin/src/main/com/looker/rtl/Transport.kt

```diff
@@ -90,20 +94,26 @@ fun <T> ok(response: SDKResponse): T {
     }
 }
 
-enum class HttpMethod(val value: io.ktor.http.HttpMethod) {
-    GET(io.ktor.http.HttpMethod.Get),
-    POST(io.ktor.http.HttpMethod.Post),
-    PUT(io.ktor.http.HttpMethod.Put),
-    DELETE(io.ktor.http.HttpMethod.Delete),
-    PATCH(io.ktor.http.HttpMethod.Patch),
-    HEAD(io.ktor.http.HttpMethod.Head)
-    // TODO: Using the ktor-client-apache may support TRACE?
+enum class HttpMethod {
+    GET,
+    POST,
+    PUT,
+    DELETE,
+    PATCH,
+    HEAD
+}
+
+enum class HttpTransports(val label: String) {
+    APACHE("Apache HTTP Client"),
+    JAVA_NET("Native Java HTTP Client"),
+    // URL_FETCH("Google App Engine HTTP Client"), TODO: App Engine support? Requires App Engine SDK.
+    // KTOR("Kotlin based HTTP Client") TODO: Add ktor transport wrapper.
```

Comment:
Is there a requirement to use `ktor`? If so I can make the wrapper class.

Apache and java.net are fairly standard.

Reply:
There's not a requirement to use `ktor` although that appears to be the more "standard" HTTP client for Kotlin. Have you looked at the dependency size impact of the Apache and/or Java HTTP clients? Is Ktor a layer on top of Java HTTP? I haven't looked since we implemented the original SDK.

Reply:
Prior to this change we were using the `OkHttp` engine but `ktor` supports several types including Apache: https://ktor.io/docs/http-client-engines.html.

In my understanding `ktor` and the Google HTTP Client do similar things in that they abstract "low-level" HTTP request/responses handling. `ktor` provides more Kotlin specific features, like coroutines, but I am not sure what else is gained. Both have Android specific transports as well if that is a requirement.

Reply:
Ah. I think @brianmadden  I talked about coroutines for `ktor` and that's why we wanted to use it, based on what he'd done in Krawler. That was a long time ago, though, so I'm not sure at this point.